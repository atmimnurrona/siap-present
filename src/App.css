@import url('https://fonts.googleapis.com/css2?family=Sora&display=swap');

*{
    font-family: 'Sora', sans-serif;
    box-sizing: border-box;
    margin: 0;
    padding: 0;
}

.spinner {
    height: 100%;
    width: 100%;
    margin: auto;
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
}

.buttonForm {
    display: flex;
    justify-content: flex-end;
}
