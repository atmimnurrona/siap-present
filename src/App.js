import './App.css';
import routes from './configs/routes'
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import { Provider } from 'react-redux';
import store from './configs/store';


import SignIn from "./pages/account/SignIn";



function App() {

  return (
    <Provider store={store}>
      <div className="App">
        <Router>
          <div>
            <Switch>
              <Route path='/' exact component={SignIn} />

              {
                routes.map((e, i) => {
                  return (
                    <Route exact={e.exact} key={i} path={e.path}>
                      {e.component}
                    </Route>
                  )
                })
              }
            </Switch>
          </div>
        </Router>
      </div>
    </Provider>
  );
}

export default App;
