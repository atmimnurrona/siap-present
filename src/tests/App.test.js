
import App from "../App"
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'
// import {expect} from 'chai';


describe('App Component ', () => {

  it('display the result', function () {
    const  tree = renderer.create(<App/>).toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('should have one provider', function () {
    const  wrapper = shallow(<App/>)
    expect(wrapper.find('Provider')).toHaveLength(1)
  });

  it('should have two div', function () {
    const wrapper = shallow(<App/>)
    expect(wrapper.find('div')).toHaveLength(2)
  });

  it('should have render', function (){
    const wrapper = shallow(<App/>)
    expect(wrapper.render()).toHaveLength(1)
  });

  it('should have one browser router', function (){
    const wrapper = shallow(<App/>)
    expect(wrapper.find('BrowserRouter')).toHaveLength(1)
  })

})