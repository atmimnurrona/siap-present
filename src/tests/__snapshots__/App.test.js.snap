// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component  display the result 1`] = `
<div
  className="App"
>
  <div>
    <div>
      <nav>
        <ul
          style={
            Object {
              "backgroundColor": "#333",
              "listStyleType": "none",
              "margin": 0,
              "overflow": "hidden",
              "padding": 0,
            }
          }
        >
          <li
            style={
              Object {
                "float": "left",
              }
            }
          >
            <a
              href="/unit"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "display": "block",
                  "padding": "14px 16px",
                  "textAlign": "center",
                  "textDecoration": "none",
                }
              }
            >
              Unit Form
            </a>
          </li>
          <li
            style={
              Object {
                "float": "left",
              }
            }
          >
            <a
              href="/units"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "display": "block",
                  "padding": "14px 16px",
                  "textAlign": "center",
                  "textDecoration": "none",
                }
              }
            >
              Unit List
            </a>
          </li>
          <li
            style={
              Object {
                "float": "left",
              }
            }
          >
            <a
              href="/items"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "display": "block",
                  "padding": "14px 16px",
                  "textAlign": "center",
                  "textDecoration": "none",
                }
              }
            >
              Item List
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div>
      Page not found!
    </div>
  </div>
</div>
`;
